src/clients/BundleDataClient/utils/FillUtils.ts

```diff
@@ -55,17 +55,18 @@ export async function verifyFillRepayment(
   matchedDeposit: DepositWithBlock,
   possibleRepaymentChainIds: number[]
 ): Promise<FillWithBlock | undefined> {
+  const updatedFill = _.cloneDeep(fill);
```

Comment:
Just refactored here. We should always be returning a clone of the fill, even if we exit early